polygeist/tools/cgeist/driver.cc

```diff
@@ -369,8 +369,13 @@ static int optimize(mlir::MLIRContext &Ctx,
     // operations to be inlined.
     if (RaiseToAffine)
       OptPM.addPass(mlir::createLowerAffinePass());
-    PM.addPass(sycl::createInlinePass(sycl::InlineMode::Simple,
-                                      /* RemoveDeadCallees */ true));
+    if (OmitOptionalMangledFunctionName) {
```

Comment:
`-no-mangled-function-name` is incompatible with inlining `SYCLMethodOps`, so should the check be this instead?
```suggestion
    if (!OmitOptionalMangledFunctionName && InlineSYCLMethodOps) {
```

Reply:
It is irrelevant whether `InlineSYCLMethodOps` is set or not, as the issue comes in an earlier step: when building the CFG. I adapted the comment to make it more obvious.